### PR TITLE
add charts to operator

### PR DESCRIPTION
**Pull request by rcernich**
Signed-off-by: rcernich <rcernich@redhat.com>

@jwendell, @brian-avery, I don't know how to test this, but it should be "mostly" correct.

### CODE REVIEW

File: istio-operator/istio-operator.spec [hunk 1]
```diff
@@ -55,15 +62,24 @@ Istio-operator is a kubernetes operator to manage the lifecycle of Istio.
...
 %build
 cd OPERATOR
 export GOPATH=$(pwd):%{gopath}
 cd src/github.com/maistra/istio-operator/tmp/build/
 ./build.sh
 
+cd OPERATOR
+cp -r src/github.com/maistra/istio/install/kubernetes/helm/ src/github.com/maistra/istio-operator/tmp/build/_output
+cd src/github.com/maistra/istio-operator/
+MAISTRA_VERSION=0.10.0 SOURCE_DIR=. HELM_DIR=src/github.com/maistra/istio-operator/tmp/build/_output/helm ./tmp/build/patch-charts.sh
```
Comment (brian-avery) [inline istio-operator/istio-operator.spec]: Can you pick up MAISTRA_VERSION from the Version variable (line 30)?

File: istio-operator/istio-operator.spec [hunk 2]
```diff
@@ -78,10 +94,20 @@ cd tmp/_output/bin/
     cp -pav stripped/istio-operator $RPM_BUILD_ROOT%{_bindir}/
 %endif
 
+# install the charts
+mkdir -p $RPM_BUILD_ROOT%{_sysconfdir}/istio-operator/1.1.0
```
Comment (brian-avery) [inline istio-operator/istio-operator.spec]: Can we move the 1.1.0 stuff to a global?

**Review by brian-avery — APPROVED** [review]
Added a few comments. Other than that, LGTM

File: istio-operator/istio-operator.spec (hunk 2 above)
Comment (rcernich) [reply istio-operator/istio-operator.spec]: Eventually.  It's currently hard-coded in the operator itself.

File: istio-operator/istio-operator.spec (hunk 1 above)
Comment (rcernich) [reply istio-operator/istio-operator.spec]: updated.  please check.  i'm assuming that it should be `%{VERSION}`.

File: istio-operator/istio-operator.spec [hunk 3]
```diff
@@ -55,15 +62,24 @@ Istio-operator is a kubernetes operator to manage the lifecycle of Istio.
...
 %build
 cd OPERATOR
 export GOPATH=$(pwd):%{gopath}
 cd src/github.com/maistra/istio-operator/tmp/build/
 ./build.sh
 
+cd OPERATOR
+cp -r src/github.com/maistra/istio/install/kubernetes/helm/ src/github.com/maistra/istio-operator/tmp/build/_output
+cd src/github.com/maistra/istio-operator/
+MAISTRA_VERSION=%{VERSION} SOURCE_DIR=. HELM_DIR=src/github.com/maistra/istio-operator/tmp/build/_output/helm ./tmp/build/patch-charts.sh
```
Comment (rcernich) [inline istio-operator/istio-operator.spec]: To make a review easier, here's the new "version." ;)

File: istio-operator/istio-operator.spec (hunk 2 above)
Comment (brian-avery) [reply istio-operator/istio-operator.spec]: By global, I meant global variable. Set it in one place in the file and we can update it on releases.

File: istio-operator/istio-operator.spec [hunk 4]
```diff
@@ -31,9 +37,10 @@ Version:        0.10.0
 Release:        1%{?dist}
 Summary:        A Kubernetes operator to manage Istio.
 License:        ASL 2.0
-URL:            https://%{provider_prefix}
+URL:            https://%{provider_prefix}/%{repo}
 
-Source0:        https://%{provider_prefix}/archive/%{git_commit}/%{repo}-%{git_commit}.tar.gz
+Source0:        https://%{provider_prefix}/%{repo}/archive/%{git_commit}/%{repo}-%{git_commit}.tar.gz
+Source1:        https://%{provider_prefix}/%{chart_repo}/archive/%{charts_git_commit}/%{chart_repo}-%{charts_git_commit}.tar.gz
```
Comment (brian-avery) [inline istio-operator/istio-operator.spec]: charts_repo

File: istio-operator/istio-operator.spec (hunk 1 above)
Comment (brian-avery) [reply istio-operator/istio-operator.spec]: Thanks

File: istio-operator/istio-operator.spec [hunk 5]
```diff
@@ -78,10 +96,20 @@ cd tmp/_output/bin/
     cp -pav stripped/istio-operator $RPM_BUILD_ROOT%{_bindir}/
 %endif
 
+# install the charts
+mkdir -p $RPM_BUILD_ROOT%{_sysconfdir}/istio-operator/%{charts_version}
+cd OPERATOR/src/github.com/maistra/istio-operator/tmp/build/
+cp -rpav _output/helm/ $RPM_BUILD_ROOT%{_sysconfdir}/istio-operator/%{charts_version}
```
Comment (rcernich) [inline istio-operator/istio-operator.spec]: Is this what you were thinking, and is it correct?

Comment (brian-avery) [reply istio-operator/istio-operator.spec]: Yep. Exactly what I was thinking. Have you tested it? I can pull it and run it if you want.

Comment (brian-avery) [reply istio-operator/istio-operator.spec]: Thanks

Comment (rcernich) [reply istio-operator/istio-operator.spec]: That would be great if you could test it really quickly.  TBH, I have no idea what I'm doing, nor how to test.  (I'm sure that fills you with confidence. ;))

File: istio-operator/istio-operator.spec [hunk 6]
```diff
@@ -31,9 +39,10 @@ Version:        0.10.0
 Release:        1%{?dist}
```
Comment (rcernich) [inline istio-operator/istio-operator.spec:42]: Do I need to change this from `1` to `2`?

File: istio-operator/istio-operator.spec (hunk 5 above)
Comment (brian-avery) [reply istio-operator/istio-operator.spec]: ./update.sh

followed by
fedpkg --release el7 srpm

followed by 
copr build @bavery/istio istio-operator-dda6d0aaada2d5bdaf82a6f1ce316877673c829a.tar.gz 

Tested it and it works fine.

File: istio-operator/istio-operator.spec (hunk 6 above)
Comment (brian-avery) [reply istio-operator/istio-operator.spec:42]: This isn't released yet, so I wouldn't worry about it. This should be incremented with rebuilds.

Comment (rcernich) [reply istio-operator/istio-operator.spec:42]: let me update the revision history then

Comment (rcernich) [reply istio-operator/istio-operator.spec:42]: updated.

**Comment by rcernich** [comment]
added flag for differentiating community/product builds

**Comment by rcernich** [comment]
This should be good to go.

File: istio-operator/istio-operator.spec [hunk 7]
```diff
@@ -13,15 +13,26 @@
...
+%global git_commit dda6d0aaada2d5bdaf82a6f1ce316877673c829a
 %global git_shortcommit  %(c=%{git_commit}; echo ${c:0:7})
 
 %global provider        github
 %global provider_tld    com
 %global project         maistra
 %global repo            istio-operator
+
+# charts
+%global charts_git_commit 4408997686d86a157f006403acfd46508d166f8c
```
Comment (brian-avery) [inline istio-operator/istio-operator.spec]: This will need to be updated for 1.1.1

File: istio-operator/istio-operator.spec [hunk 8]
```diff
@@ -13,15 +13,26 @@
...
+%global chargs_git_shortcommit  %(c=%{charts_git_commit}; echo ${c:0:7})
+
+%global charts_repo      istio
+%global charts_version   1.1.0
```
Comment (brian-avery) [inline istio-operator/istio-operator.spec:29]: 1.1.1

**Review by brian-avery — APPROVED** [review]

File: istio-operator/istio-operator.spec (hunk 8 above)
Comment (rcernich) [reply istio-operator/istio-operator.spec:29]: this path is hard coded in the operator itself.  can we leave it as 1.1.0 for now and revisit in a following sprint?  we need a better design than what we have.

**Comment by brian-avery** [comment]
Sounds good to me.

**Pull request merged**